src/middleware/web-outgoing.ts

```diff
@@ -45,7 +45,7 @@ const setRedirectHostRewrite = defineProxyOutgoingMiddleware(
       if (options.hostRewrite) {
         u.host = options.hostRewrite;
       } else if (options.autoRewrite) {
-        u.host = req.headers.host;
+        u.host = req.headers.host ?? "undefined";
```

Comment:
This is to keep the behavior. Maybe this should be:
```suggestion
        if (req.headers.host) {
          u.host = req.headers.host;
        }
```